app/contacts/contact_ftp.py

```diff
@@ -0,0 +1,219 @@
+                    bytes_obj += data
+
+            if re.match(r"^Alive\.txt$", name[len(name) - 1]):
+                file = True
+                profile = json.loads(bytes_obj.decode())
+                paw, response = await r_class.create_response(profile)
+                success = r_class.write_response_file(paw, response)
+                if not success:
+                    self.logger.debug("ERROR: Failed to create response")
+
+            if re.match(r"^Payload\.txt$", name[len(name) - 1]):
+                p_load = True
+                profile = json.loads(bytes_obj.decode())
+                file_path, contents, display_name = await r_class.get_payload_file(profile)
+                if file_path is not None:
+                    r_class.write_file(profile.get('paw'), profile.get('file'), str(contents))
+
+            if not file and not p_load:
+                paw = name[len(name) - 2]
+                filename = name[len(name) - 1]
+                await r_class.submit_uploaded_file(paw, filename, bytes_obj)
+
+            connection.response("226", "data transfer done")
+            del stream
+            return True
+
+        real_path, virtual_path = self.get_paths(connection, rest)
+        name = str(virtual_path).split("/")
+        self.logger.debug("File: " + str(name[len(name) - 1]) + " received")
+        r_class = CalderaServer(self.contact_svc, self.file_svc, self.logger, self.home, self.directory)
+
+        if await connection.path_io.is_dir(real_path.parent):
+            coro = stor_worker(self, connection, rest)
+            task = asyncio.create_task(coro)
+            connection.extra_workers.add(task)
+            code, info = "150", "data transfer started"
+        else:
+            code, info = "550", "path unreachable"
+        connection.response(code, info)
+        return True
+
+
+class CalderaServer:
+    def __init__(self, contact, file, log, h, d):
+        self.contact_svc = contact
+        self.file_svc = file
+        self.logger = log
+        self.home = h
+        self.directory = d
+
+    async def create_response(self, profile):
+        paw = profile.get('paw')
+        profile['paw'] = paw
+        profile['contact'] = profile.get('contact', 'ftp')
+        agent, instructions = await self.contact_svc.handle_heartbeat(**profile)
+        response = dict(paw=agent.paw,
+                        sleep=await agent.calculate_sleep(),
+                        watchdog=agent.watchdog,
+                        instructions=json.dumps([json.dumps(i.display) for i in instructions]))
+        if agent.pending_contact != agent.contact:
+            response['new_contact'] = agent.pending_contact
+            self.logger.debug('Sending agent instructions to switch from C2 channel %s to %s'
+                              % (agent.contact, agent.pending_contact))
+        return paw, response
+
+    def write_response_file(self, paw, response):
```

Comment:
since there's also another function called `write_file`, it'll be helpful to make the names more specific so we know the purpose of each file being written (e.g. `write_beacon_response_file` and `write_payload_file`)